### ISSUE TITLE
HTTP - GET - invalid gzip

### ISSUE THREAD

**Issue by 2xyo** [Cisco-Talos/file2pcap#8]
```console
$ wget http://www.eicar.org/download/eicar.com.txt
$ file eicar.com.txt
eicar.com.txt: EICAR virus test files
$./file2pcap --srcip 1.1.1.1 --dstip 2.2.2.2 --block-size 1200 -mh --http-encoder 1 -o /tmp/payload-1200_http-GET-gzip.pcap eicar.com.txt
Writing to /tmp/payload-1200_http-GET-gzip.pcap
$ tshark -r /tmp/payload-1200_http-GET-gzip.pcap -Y 'http.response'  -T fields -e text -e data
Timestamps,HTTP/1.1 200 Ok\r\n,\r\n,Content-encoded entity body (gzip): 80 bytes [Error: Decompression failed]	1f8b0800000000000003008b30f5570c5075700c88368909888a3035d10888d3347776d634af5571f574760cd20d0e71f473710c72d175f40bf10cf30c0a0dd60d710d0ed175f3f4715554f1d0f6d002,00
$ gzip -c -k ./eicar.com.txt |xxd -p -c 1024
1f8b0808aac74f5e000365696361722e636f6d2e747874008b30f5570c5075700c88368909888a3035d10888d3347776d634af5571f574760cd20d0e71f473710c72d175f40bf10cf30c0a0dd60d710d0ed175f3f4715554f1d0f6d002003ccf516844000000
```

Ok with other encoders:
```console
$ ./file2pcap --srcip 1.1.1.1 --dstip 2.2.2.2 --block-size 1200 -mh --http-encoder 0 -o /tmp/payload-1200_http-GET-defaut.pcap eicar.com.txt
$ tshark -r /tmp/payload-1200_http-GET-defaut.pcap -Y 'http.response'  -T fields -e text -e data
Timestamps,HTTP/1.1 200 Ok\r\n,\r\n	58354f2150254041505b345c505a58353428505e2937434329377d2445494341522d5354414e444152442d414e544956495255532d544553542d46494c452124482b482a
$ echo 58354f2150254041505b345c505a58353428505e2937434329377d2445494341522d5354414e444152442d414e544956495255532d544553542d46494c452124482b482a| xxd -r -p
X5O!P%@AP[4\PZX54(P^)7CC)7}$EICAR-STANDARD-ANTIVIRUS-TEST-FILE!$H+H*

$./file2pcap --srcip 1.1.1.1 --dstip 2.2.2.2 --block-size 1200 -mh --http-encoder 2 -o /tmp/payload-1200_http-GET-chunked.pcap eicar.com.txt
Writing to /tmp/payload-1200_http-GET-chunked.pcap
$ tshark -r /tmp/payload-1200_http-GET-chunked.pcap -Y 'http.response'  -T fields -e text -e data
Timestamps,HTTP/1.1 200 Ok\r\n,\r\n,HTTP chunked response,Data chunk (68 octets),End of chunked encoding,\r\n,X5O!P%@AP[4\PZX54(P^)7CC)7}$EICAR-STANDARD-ANTIVIRUS-TEST-FILE!$H+H*	58354f2150254041505b345c505a58353428505e2937434329377d2445494341522d5354414e444152442d414e544956495255532d544553542d46494c452124482b482a

$./file2pcap --srcip 1.1.1.1 --dstip 2.2.2.2 --block-size 1200 -mh --http-encoder 3 -o /tmp/payload-1200_http-GET-gzip-chunked.pcap eicar.com.txt
Writing to /tmp/payload-1200_http-GET-gzip-chunked.pcap
$ tshark -r /tmp/payload-1200_http-GET-gzip-chunked.pcap -Y 'http.response'  -T fields -e text -e data
Timestamps,HTTP/1.1 200 Ok\r\n,\r\n,HTTP chunked response,Data chunk (10 octets),Data chunk (70 octets),Data chunk (8 octets),End of chunked encoding,\r\n,Content-encoded entity body (gzip): 88 bytes -> 68 bytes,X5O!P%@AP[4\PZX54(P^)7CC)7}$EICAR-STANDARD-ANTIVIRUS-TEST-FILE!$H+H*	1f8b0800000000000003,8b30f5570c5075700c88368909888a3035d10888d3347776d634af5571f574760cd20d0e71f473710c72d175f40bf10cf30c0a0dd60d710d0ed175f3f4715554f1d0f6d00200,3ccf516844000000

```

**Comment by Talos-Martin** [comment]
Issue confirmed and tracked down. Fixed now in my version. File2pcap will have it fixed in the 1.30 release.

**Comment by Talos-Martin** [comment]
Fixed in 1.30.
